pkg/kubelet/prober/prober.go

```diff
@@ -112,6 +112,9 @@ func (pb *prober) probe(ctx context.Context, probeType probeType, pod *v1.Pod, s
 	if result == probe.Warning {
 		pb.recordContainerEvent(pod, &container, v1.EventTypeWarning, events.ContainerProbeWarning, "%s probe warning: %s", probeType, output)
 		klog.V(3).InfoS("Probe succeeded with a warning", "probeType", probeType, "pod", klog.KObj(pod), "podUID", pod.UID, "containerName", container.Name, "output", output)
+	} else if result == probe.Unknown {
+		klog.V(3).InfoS("Probe resulted in unknown state", "probeType", probeType, "pod", klog.KObj(pod), "podUID", pod.UID, "containerName", container.Name)
+		pb.recordContainerEvent(pod, &container, v1.EventTypeWarning, events.ContainerProbeWarning, "%s probe resulted in unknown state: %s", probeType, output)
```

Comment:
Would it be more readable to use `switch` instead of bunch of `if` and `else if` statements?

Reply:
I've used `switch` before in this PR, but I think the `lint` build failed: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/125901/pull-kubernetes-linter-hints/1821041892498870272

Any pointer to get `linter` working in this case ?

Reply:
Hopefully that one has been resolved with latest changes.